### ISSUE TITLE
Feedreader can't read links in CDATA sections

### ISSUE THREAD

**Issue by ksl1989** [jangernert/FeedReader#815]
Sixth Tone puts its feed's links in CDATA sections. I don't know if this is allowed and it's certainly unneeded but there's no reason why Feedreader shouldn't be able to read the link in the CDATA section, and it's possible other feeds with be formatted the same way.

```
<item>
            <title><![CDATA[Tax Crackdown Shocks, Worries China’s Entertainment Industry]]></title>
            <link><![CDATA[//www.sixthtone.com/news/1003289/Tax Crackdown Shocks, Worries China’s Entertainment Industry]]></link>
            <pubDate>Dec 4, 2018</pubDate>
            <author>Yin Yijun</author>
            <category><![CDATA[RisingTones]]></category>
            <description><![CDATA[<a href="/news/1003289/Tax Crackdown Shocks, Worries China’s Entertainment Industry"><img src="/image/5/14/934.jpg" width="204" height="109" alt=""></a></br>A document indicating that artists and studios may have to pay additional taxes for the past three years has caused anxiety and backlash.]]></description>
           <type>text</type>
</item>
```
https://www.sixthtone.com/rss

**Comment by brendanlong** [comment]
I can check, but I'm pretty sure our XML parser handles CDATA correctly. The problem is that this isn't a valid URL, because it contains spaces (and commas might not be allowed either?):

```
//www.sixthtone.com/news/1003289/Tax Crackdown Shocks, Worries China’s Entertainment Industry
```

This link should be:

```
//www.sixthtone.com/news/1003289/Tax%20Crackdown%20Shocks%2C%20Worries%20China%E2%80%99s%20Entertainment%20Industry
```

(CDATA has no effect in this case because there aren't any `<>` characters)

If we don't support URL's starting with `//` I'd be willing to add that since it's part of [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2) but I don't think we want to go down the path of trying to fix invalid URL's.

**Comment by brendanlong** [comment]
Although maybe it's easy enough to find invalid characters and URL escape them? I'm guessing that's what Firefox does if you paste these links into your URL bar. Hm.

**Comment by brendanlong** [comment]
I will note that both Firefox and Feedbin can't understand the links in this feed. Firefox ignores them and Feedbin gives up and gives you a link to the top level domain:

![screen shot 2018-12-04 at 5 58 13 pm](https://user-images.githubusercontent.com/1447206/49478673-6eeb6d80-f7ee-11e8-9727-f86784dbda41.png)

**Comment by ksl1989** [comment]
My mistake, then. I compared this feed to several other feeds that were working in FeedReader and I noticed this was the only one with links in CDATA sections, so I thought this was the issue.

So it is either the `//` URLs which should be a considered a bug, as it's part of the specification, or the unescaped spaces. This wouldn't be a bug with FeedReader, but since it's easy to fix and would also make similarly formatted feeds work (supposing there are more), I think it would be a good thing to do. As you mentioned, Firefox ignores the links in the feeds, but it has no problem opening them if you paste them on the URL box.

**Comment by brendanlong** [comment]
Yeah, I think if someone makes a pull request to make this work I'd be willing to review and merge it. I'm guessing we would want to put this in the local plugin (and DecSync plugin too until we can duduplicate their shared code more).
